packages/kbn-rule-data-utils/src/technical_field_names.ts

```diff
@@ -88,6 +88,9 @@ const ALERT_EVALUATION_THRESHOLD = `${ALERT_NAMESPACE}.evaluation.threshold` as
 const ALERT_EVALUATION_VALUE = `${ALERT_NAMESPACE}.evaluation.value` as const;
 const ALERT_CONTEXT = `${ALERT_NAMESPACE}.context` as const;
 const ALERT_EVALUATION_VALUES = `${ALERT_NAMESPACE}.evaluation.values` as const;
+const ALERT_GROUPS = `${ALERT_NAMESPACE}.groups` as const;
```

Comment:
The structure of this field is similar to how the security team saves the information [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/field_maps/8.0.0/alerts.ts#L229-L238)

I would like to introduce this field in a way that we will use the same for the elasticsearch query rule. Please let me know if there is a better naming out there :)

Reply:
IMO, the names are 👍 . We'll want to get @shanisagiv1's opinion as well on the names.

However, I think we need to change the mappings... In retrospect, my suggestion that we just copy what the security solution team has done here was wrong.

With the existing mappings, it's impossible to write a filter that will match on both `kibana.alert.groups.field` and `kibana.alert.groups.value`.

For example, let's assume that we have two alerts created with the following groups:

**alert 1**
```
"kibana.alert.groups": [
    {
      "field": "host.name.keyword",
      "value": "baz"
    },
    {
      "field": "server.name.keyword",
      "value": "foo"
    }
]
```

**alert 2**
```
"kibana.alert.groups": [
    {
      "field": "host.name.keyword",
      "value": "foo"
    },
    {
      "field": "server.name.keyword",
      "value": "bar"
    }
]
```

If we then run the following ES Query which is intended to only retrieve alerts with `host.name.keyword: "foo"`, it will return both alert 1 and alert 2:

```
GET .alerts-*/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "kibana.alert.groups.field": "host.name.keyword"
          }
        },
        {
          "term": {
            "kibana.alert.groups.value": "foo"
          }
        }
      ]
    }
  }
}
```

This is because of the way that ES flattens arrays of objects. There is a good description of this behavior here: https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html#nested-arrays-flattening-objects

This leaves us with two options:
1. [A nested field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html)
2. (my recommendation) [A flattened field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/flattened.html)

The nested field type comes with a number of caveats that the docs cover well: https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html#nested-arrays-flattening-objects . Given our knowledge of how this field will be populated at the moment, we could prolly get away with using a nested field type. However, it's hard to predict the future and know whether these limitations will become relevant in the future.

Therefore, I think the flattened field type makes more sense because it doesn't have many of the caveats that the nested field type has. It does have some, for example all values are treated like a keyword, but for this particular situation, I don't foresee those as being relevant.

Reply:
@simianhacker what do you think is the best format of saving group information based on what @kobelb said?

Reply:
@maryam-saeidi I agree with @kobelb, I think a flattened field would be better.

Reply:
@kobelb One question:
Is the downside of a nested field type the fact that we need to be aware of the field type and use a nested query for this field or something else? I wasn't sure which part of the link that you shared was relevant to the caveats (Also, not sure how this will work with the unified search bar)

Reply:
@maryam-saeidi that's definitely one of the big downsides. The other caveat I'm concerned about here is [Limits on nested mappings and objects](https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html#_limits_on_nested_mappings_and_objects). 

With your all's usage of the `kibana.alert.groups` field, we shouldn't end up with a ton of documents because we'll just have an item per "group by key". There probably won't be a ton of "group by keys" with your all's usage, but it could become an issue for some future usage. I'd prefer we not have to think about having to limit the items in this array if we don't have to.

Reply:
Hey,
re naming - `kibana.alert.group.`x or `kibana.alert.groupby.` make sense to me

2 questions:
- How this field interacts with the context syntax? `${context.alert.kibana.alert.group}`will work? if there are multiple group_by fields, it lists all as an array? will autocomplete list `context.alert.kibana.alert.group.host.name` or will it just list `context.alert.kibana.alert.group`?
- how it works in the alert table? users have to specify the exact column?